include/swift/AST/DiagnosticsSema.def

```diff
@@ -2981,6 +2981,8 @@ ERROR(override_rethrows_with_non_rethrows,none,
       "be 'rethrows'", (bool))
 ERROR(rethrows_without_throwing_parameter,none,
       "'rethrows' function must take a throwing function argument", ())
+ERROR(rethrows_attr_on_non_protocol,none,
```

Comment:
This shouldn't be necessary because you annotated the attribute with OnProtocol in Attrs.def

Reply:
so that means the other code:
```
void AttributeChecker::visitAtRethrowsAttr(AtRethrowsAttr *attr) {
  if (isa<ProtocolDecl>(D)) {
    return;
  }

  diagnose(attr->getLocation(), diag::rethrows_attr_on_non_protocol);
  attr->setInvalid();
}
```
should change to this?
```
void AttributeChecker::visitAtRethrowsAttr(AtRethrowsAttr *attr) { }
```